lib/stateView/document/query/jsonSchema.js

```diff
@@ -0,0 +1,223 @@
+module.exports = {
+  type: 'object',
+  definitions: {
+    field: {
+      $id: 'field',
+      type: 'string',
+      minLength: 1,
+      maxLength: 255,
+      pattern: '^(\\$id|\\$userId|[a-zA-Z0-9-_.]+)$',
```

Comment:
We need more strict regexp to prevent using dots at the beginning, at the end and without something in between.